content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -147,6 +147,30 @@ const xprisma = prisma.$extends({
 })
 ```
 
+### Mutate the result of a query
+
+You can use `await` and then mutate the result of the `query` promise, because its value is type-safe. TypeScript catches any unsafe mutations on the object.
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      async findFirst({ model, operation, args, query }) {
+        const user = await query(args)
+        
+        if (user.password !== undefined) {
+          user.password = '******'
+        }
+
+        return user
```

Comment:
Interesting that we can either return an unresolved "Promise" or the data directly - is that correct?

Reply:
In which case is the promise unresolved? Do mean that sometimes, we just `return query(args)` without marking the function `async` and in other cases we do like here `const user = await query(args)` and then return that value?

Reply:
@janpio @millsp  I'd like to merge this PR. Are you both happy that this conversation is resolved?

Reply:
Yep.